src/ansiblelint/utils.py

```diff
@@ -759,6 +760,36 @@ def construct_mapping(
     return data
 
 
+def iter_tasks_in_file(lintable: Lintable, rule_id: str):
```

Comment:
This method yields but the signature does not report that. I suspect something like this should do the trick https://stackoverflow.com/a/57916822/99834

In fact I was curious why our linters did not catch this, probably badly configured.

Reply:
Fixed.

The linter did catch it, but I hadn't configured pre-commit yet.

I just ran `pre-commit install` to make sure I don't push changes like this again.